### ISSUE TITLE
Stripes on KITTI results 

### ISSUE THREAD

**Issue by waychin-weiqin** [youmi-zym/AcfNet#3]
Hi, 

First of all great work. Really impressed with your work. But I am curious to see the lines/stripes that appear on your KITTI error maps (see below for example). May I know what could be the reason for that occurrence? 

![image](https://user-images.githubusercontent.com/33308266/85087282-6ddb2f00-b220-11ea-8cc5-fe2b9cbad26e.png)

**Comment by youmi-zym** [comment]
Hi @SamChuah,
Thanks for your interest in our work. With regard to the KITTI error maps you cropped, all errors are very close actually, about 0.01 to 0.5 pixels. 
With regard to the behind reason, I think it's an incompatible problem between our unimodal distribution supervision and soft-argmin disparity regression function. As we just want only one peak appears at the ground-truth disparity, but soft-argmin considering all disparity indexes when regression, so some kind of lines/stripes appears. 
To alleviate this problem, we also provide a new regression method: local soft arg-min [here](https://github.com/DeepMotionAIResearch/DenseMatchingBenchmark/blob/master/dmb/modeling/stereo/disp_predictors/local_soft_argmin.py), you can follow the [guidance](https://github.com/DeepMotionAIResearch/DenseMatchingBenchmark/blob/master/configs/AcfNet/ResultOfAcfNet.md#how-to) to use it.

**Comment by waychin-weiqin** [comment]
In your work you implemented soft-argmin to obtain the disparity value from the cost volume. I am assuming if the network is trained using soft-argmax, it will learn to predict similarity scores instead of matching costs and will give similar results?

**Comment by youmi-zym** [comment]
For the stereo matching task, cost volume will be converted into the similarity scores through softmax. The problem I'm talking about is the **soft-argmin considering all disparity indexes** (i.e., 0, 1, 2, ..., MaxDisparity) when regression, which cannot precisely pick out the peak in the cost distribution. To reduce the influence of some useless disparity indexes, we try to replace soft-argmin with local soft-argmin.
